### PR TITLE
Fix Mac OS SDK 14 deprecated items or suppress warnings

### PR DESCRIPTION
**Pull request by aefimov3**
## What does this PR do?

Change / mask Mac OS SDK 14 deprecated calls.

## How was this PR tested?

Local test with building mac binaries.

### CODE REVIEW

File: Code/Framework/AzQtComponents/AzQtComponents/Utilities/ScreenGrabber_mac.mm [hunk 1]
```diff
@@ -57,8 +57,10 @@
         QRect region({}, m_size);
         region.moveCenter(point);
         CGRect bounds = CGRectMake(region.x(), region.y(), region.width(), region.height());
-
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
         CGImageRef cgImage = CGWindowListCreateImageFromArray(bounds, windows, kCGWindowImageNominalResolution);
```
Comment (aefimov3) [inline Code/Framework/AzQtComponents/AzQtComponents/Utilities/ScreenGrabber_mac.mm:65]: This one is not nice, but I see no quick substitute.

File: Gems/Atom/RHI/Metal/Code/Source/RHI/ArgumentBuffer.cpp [hunk 2]
```diff
@@ -138,7 +138,7 @@ namespace AZ
                 MTLArgumentDescriptor* samplerArgDescriptor = [[[MTLArgumentDescriptor alloc] init] autorelease];
                 samplerArgDescriptor.dataType = MTLDataTypeSampler;
                 samplerArgDescriptor.index = shaderInputSampler.m_registerId;
-                samplerArgDescriptor.access = MTLArgumentAccessReadOnly;
+                samplerArgDescriptor.access = MTLBindingAccessReadOnly;
```
Comment (aefimov3) [inline Gems/Atom/RHI/Metal/Code/Source/RHI/ArgumentBuffer.cpp:141]: Everything else is pretty obvious.

File: Code/Framework/AzQtComponents/AzQtComponents/Utilities/ScreenGrabber_mac.mm [hunk 3]
```diff
@@ -57,8 +57,10 @@
...
+#pragma clang diagnostic pop
         CFRelease(windows);
```
Comment (Twolewis) [inline Code/Framework/AzQtComponents/AzQtComponents/Utilities/ScreenGrabber_mac.mm]: Wrap this in a #CARBONATED so we can track it.  Also, please make a GHI for it (and link in this PR) so the O3DE folks can tackle

**Review by Twolewis — CHANGES_REQUESTED** [review]

File: Code/Framework/AzQtComponents/AzQtComponents/Utilities/ScreenGrabber_mac.mm (hunk 3 above)
Comment (aefimov3) [reply Code/Framework/AzQtComponents/AzQtComponents/Utilities/ScreenGrabber_mac.mm]: Added CARBONATED fence, created GHI:
https://github.com/o3de/o3de/issues/17928

**Review by Twolewis — APPROVED** [review]

**Pull request merged**